lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -444,19 +465,49 @@ singleAddressFromKey _ xPub = Address $ BL.toStrict $ runPut $ do
 isolatePut :: Int -> Put -> Put
 isolatePut l x = do
     let bs = runPut x
-    if BL.length bs == (fromIntegral l)
+    if BL.length bs == convert l
     then putLazyByteString bs
     else fail $ "length was "
         ++ show (BL.length bs)
         ++ ", but expected to be "
         ++ (show l)
+  where
+    convert :: Int -> Int64
+    convert = fromIntegral
 
 withSizeHeader16be :: Put -> Put
 withSizeHeader16be x = do
     let bs = BL.toStrict $ runPut x
-    putWord16be (fromIntegral $ BS.length bs)
+    putWord16be (toEnum $ BS.length bs)
     putByteString bs
 
+-- | Like 'fromIntegral' but throws an exception if the conversion would have
+-- caused overflow.
+--
+-- > f = unsafeFromIntegral @Int @Word8
+-- > f 0
+-- 0
+-- > f 255
+-- 255
+-- > f 256
+-- *** Exception: unsafeFromIntegral: cannot convert (256 :: Int) to Word8
+unsafeFromIntegral
+    :: forall i j . (Show i, Typeable i, Integral i)
+    => (Typeable j, Bounded j, Integral j)
+    => i
+    -> j
+unsafeFromIntegral i
```

Comment:
This is redundant with `toEnum . fromEnum`

Reply:
Do you object to the helper `unsafeFromIntegral = toEnum . fromEnum`?

Reply:
Hmm.. yeah. The name would be rather misleading :|